src/lib.rs

```diff
@@ -2,35 +2,63 @@ use proc_macro::TokenStream;
 use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
 use syn::{
-    parse_macro_input, punctuated::Punctuated, token::Comma, Data, DeriveInput, Field, Fields,
+    parse_macro_input, punctuated::Punctuated, token::Comma, Data, DataEnum, DataStruct,
+    DeriveInput, Field, Fields, Ident, Variant,
 };
 
 #[proc_macro_derive(Serialize)]
 pub fn derive_serialize(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
 
-    let struct_name = input.ident;
+    let type_name = input.ident;
 
-    let struct_fields = get_struct_fields(&input.data);
+    let expanded = match input.data {
+        Data::Struct(ref data_struct) => {
+            let struct_fields = get_struct_fields(data_struct);
 
-    let it = struct_fields.iter().map(|field| {
-        let name = &field.ident;
-        let keyword = to_edn_keyword(format!("{}", quote! {#name}));
-        quote! {
-            format!("{} {}, ", #keyword, self.#name.serialize())
+            let it = struct_fields.iter().map(|field| {
+                let name = &field.ident;
+                let keyword = to_edn_keyword(format!("{}", quote! {#name}));
+                quote! {
+                    format!("{} {}, ", #keyword, self.#name.serialize())
+                }
+            });
+
+            quote! {
+                impl edn_rs::Serialize for #type_name {
+                    fn serialize(self) -> String {
+                        let mut s = String::new();
+                        s.push_str("{ ");
+                        #(s.push_str(&#it);)*
+                        s.push_str("}");
+                        s
+                    }
+                }
+            }
         }
-    });
-
-    let expanded = quote! {
-        impl edn_rs::Serialize for #struct_name {
-            fn serialize(self) -> String {
-                let mut s = String::new();
-                s.push_str("{ ");
-                #(s.push_str(&#it);)*
-                s.push_str("}");
-                s
+        Data::Enum(ref data_enum) => {
+            let enum_variants = get_enum_variants(data_enum);
+
+            let it = enum_variants.iter().map(|variant| {
+                let name = &variant.ident;
+                let keyword =
+                    to_edn_keyword(format!("{}___{}", quote! {#type_name}, quote! {#name}));
```

Comment:
we don't need to run the `___` here, maybe we could just send `/`  here 🤔.
The `maybe_replace` I used in the other crate has a small performance advantage, what do you think of using it in `to_edn_keyword`?

Reply:
I guess you are right, we can avoid unnecessary replaces by sending the `/` directly.

About the maybe replace, I will create an issue so that we can test this :slightly_smiling_face:

Reply:
Oh shit I merged before changing the `/` :facepalm: 
I will push the fix on master!

Reply:
Commit: https://github.com/otaviopace/edn-derive/commit/91f877fcb7d14f6e2b7cbb6f2813399a14965896
Issue: https://github.com/otaviopace/edn-derive/issues/11

Reply:
I mean, the performance will probably be better, but I don't know if it will affect that much since this probably runs one for each field of each structure on a code base (which I guess can be high, but usually for most cases won't probably pass 10 types).